service/storage_service.cc

```diff
@@ -1271,63 +1270,54 @@ class join_node_rpc_handshaker : public service::group0_handshaker {
     }
 };
 
-future<> storage_service::raft_initialize_discovery_leader(const join_node_request_params& params) {
-    // No other server can become a member of the cluster until
-    // we finish adding ourselves. This is ensured by
-    // waiting in join_node_request_handler for
-    // _topology_state_machine._topology.normal_nodes to be not empty.
-    // We cannot mark ourselves as dead during this process.
-    // This means there is no valid way we can lose quorum here,
-    // but some subsystems may just become unreasonably slow for various reasons,
-    // so we nonetheless use raft_timeout{} here.
+future<> storage_service::raft_initialize_discovery_leader(const join_node_request_params& params, raft::server_id id) {
+    if (params.replaced_id.has_value()) {
+        throw std::runtime_error(::format("Cannot perform a replace operation because this is the first node in the cluster"));
+    }
 
-    if (_topology_state_machine._topology.is_empty()) {
-        co_await _group0->group0_server_with_timeouts().read_barrier(&_group0_as, raft_timeout{});
+    if (params.num_tokens == 0 && params.tokens_string.empty()) {
+        throw std::runtime_error("Cannot start the first node in the cluster as zero-token");
     }
 
-    while (_topology_state_machine._topology.is_empty()) {
-        if (params.replaced_id.has_value()) {
```

Comment:
mixing indentation change with everything else makes the change harder to review